src/fmcore/prompt_tuner/dspy/utils/dspy_utils.py

```diff
@@ -0,0 +1,252 @@
+import dspy
+import pandas as pd
+
+from langchain_core.prompts.chat import ChatPromptTemplate
+from typing import Callable, Dict, List
+
+from fmcore.prompt_tuner.dspy.datasets.base_dataset import DspyDataset
+from fmcore.prompt_tuner.dspy.lm_adapters.dspy_adapter import DSPyLLMAdapter
+from fmcore.prompt_tuner.evaluator import BaseEvaluator
+from fmcore.prompt_tuner.types.prompt_tuner_types import PromptConfig, PromptEvaluationResult
+from fmcore.llm.types.llm_types import LLMConfig
+from fmcore.types.enums.dataset_enums import DatasetType
+
+
+class DSPyUtils:
+    """
+    Utility class for working with DSPy in prompt tuning and optimization tasks.
+
+    This class provides static methods for setting up and configuring various components
+    of the DSPy framework, including optimizers, datasets, signatures, modules, and evaluation
+    functions. It serves as a bridge between the fmcore configuration objects and DSPy's
+    expected interfaces.
+    """
+
+    @staticmethod
+    def create_dspy_lm(llm_config: LLMConfig):
```

Comment:
looks like u are using adapter directly in different places.. is this method required